File: CS/RichEditAPISample/CodeExamples/Comments.cs
using DevExpress.XtraRichEdit.API.Native;
using System;

namespace RichEditAPISample.CodeExamples {
    class CommentsActions    {

        static void CreateComment(Document document)
        {
            #region #CreateComment
            document.LoadDocument("Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml);
            DocumentRange docRange = document.Paragraphs[2].Range;
            string commentAuthor = "Johnson Alphonso D";
            document.Comments.Create(docRange, commentAuthor, DateTime.Now);
            #endregion #CreateComment
        }

        static void CreateNestedComment(Document document) {
            #region #CreateNestedComment
            document.LoadDocument("Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml);
            if (document.Comments.Count > 0) {
                DocumentRange[] resRanges = document.FindAll("trump", SearchOptions.None, document.Comments[1].Range);
                if (resRanges.Length > 0) {
                    Comment newComment = document.Comments.Create("Vicars Anny", document.Comments[1]);
                    newComment.Date = DateTime.Now;
                }
            }
            #endregion #CreateNestedComment
        }

        static void DeleteComment(Document document)
        {
            #region #DeleteComment
            document.LoadDocument("Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml);
            if (document.Comments.Count > 0)
            {
                document.Comments.Remove(document.Comments[0]);
            }
            #endregion #DeleteComment
        }

        static void EditCommentProperties(Document document)
        {
            #region #EditCommentProperties
            document.LoadDocument("Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml);
            int commentCount = document.Comments.Count;
            if (commentCount > 0)
            {
                document.BeginUpdate();
                DevExpress.XtraRichEdit.API.Native.Comment comment = document.Comments[document.Comments.Count - 1];
                comment.Name = "New Name";
                comment.Date = DateTime.Now;
                comment.Author = "New Author";
                document.EndUpdate();
            }
            #endregion #EditCommentProperties
        }

        static void EditCommentContent(Document document)
        {
            #region #EditCommentContent
            document.LoadDocument("Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml);
            int commentCount = document.Comments.Count;
            if (commentCount > 0)
            {
                DevExpress.XtraRichEdit.API.Native.Comment comment = document.Comments[document.Comments.Count - 1];
                if (comment != null)
                {
                    SubDocument commentDocument = comment.BeginUpdate();
                    commentDocument.InsertText(commentDocument.CreatePosition(0), "some text");
                    commentDocument.Tables.Create(commentDocument.CreatePosition(9), 5, 4);
                    comment.EndUpdate(commentDocument);
                }
            }
            #endregion #EditCommentContent
        }
    }
}

File: VB/RichEditAPISample/CodeExamples/Comments.vb
Imports DevExpress.XtraRichEdit.API.Native
Imports System

Namespace RichEditAPISample.CodeExamples

    Friend Class CommentsActions

        Private Shared Sub CreateComment(ByVal document As DevExpress.XtraRichEdit.API.Native.Document)
#Region "#CreateComment"
            document.LoadDocument("Documents//Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            Dim docRange As DevExpress.XtraRichEdit.API.Native.DocumentRange = document.Paragraphs(CInt((2))).Range
            Dim commentAuthor As String = "Johnson Alphonso D"
            document.Comments.Create(docRange, commentAuthor, System.DateTime.Now)
#End Region  ' #CreateComment
        End Sub

        Private Shared Sub CreateNestedComment(ByVal document As DevExpress.XtraRichEdit.API.Native.Document)
#Region "#CreateNestedComment"
            document.LoadDocument("Documents//Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            If document.Comments.Count > 0 Then
                Dim resRanges As DevExpress.XtraRichEdit.API.Native.DocumentRange() = document.FindAll("trump", DevExpress.XtraRichEdit.API.Native.SearchOptions.None, document.Comments(CInt((1))).Range)
                If resRanges.Length > 0 Then
                    Dim newComment As DevExpress.XtraRichEdit.API.Native.Comment = document.Comments.Create("Vicars Anny", document.Comments(1))
                    newComment.[Date] = System.DateTime.Now
                End If
            End If
#End Region  ' #CreateNestedComment
        End Sub

        Private Shared Sub DeleteComment(ByVal document As DevExpress.XtraRichEdit.API.Native.Document)
#Region "#DeleteComment"
            document.LoadDocument("Documents//Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            If document.Comments.Count > 0 Then
                document.Comments.Remove(document.Comments(0))
            End If
#End Region  ' #DeleteComment
        End Sub

        Private Shared Sub EditCommentProperties(ByVal document As DevExpress.XtraRichEdit.API.Native.Document)
#Region "#EditCommentProperties"
            document.LoadDocument("Documents//Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            Dim commentCount As Integer = document.Comments.Count
            If commentCount > 0 Then
                document.BeginUpdate()
                Dim comment As DevExpress.XtraRichEdit.API.Native.Comment = document.Comments(document.Comments.Count - 1)
                comment.Name = "New Name"
                comment.[Date] = System.DateTime.Now
                comment.Author = "New Author"
                document.EndUpdate()
            End If
#End Region  ' #EditCommentProperties
        End Sub

        Private Shared Sub EditCommentContent(ByVal document As DevExpress.XtraRichEdit.API.Native.Document)
#Region "#EditCommentContent"
            document.LoadDocument("Documents//Grimm.docx", DevExpress.XtraRichEdit.DocumentFormat.OpenXml)
            Dim commentCount As Integer = document.Comments.Count
            If commentCount > 0 Then
                Dim comment As DevExpress.XtraRichEdit.API.Native.Comment = document.Comments(document.Comments.Count - 1)
                If comment IsNot Nothing Then
                    Dim commentDocument As DevExpress.XtraRichEdit.API.Native.SubDocument = comment.BeginUpdate()
                    commentDocument.InsertText(commentDocument.CreatePosition(0), "some text")
                    commentDocument.Tables.Create(commentDocument.CreatePosition(9), 5, 4)
                    comment.EndUpdate(commentDocument)
                End If
            End If
#End Region  ' #EditCommentContent
        End Sub
    End Class
End Namespace
